### DISCUSSION TITLE
Append animation files in one

### DISCUSSION THREAD

**Discussion by kalthwaini** [donmccurdy/glTF-Transform#580] [Q&A] [answered]
Am try to append multiple animation files into one longer file. All files are bones only (skeletons).

after, i need to apply a character model to the longer file as well.

is such a need approachable with gltf-transform?

**Comment by donmccurdy** [comment, marked as answer]
This is certainly possible to implement, but will require writing a decent amount of code. I would recommend looking at the documentation for the [Animation](https://gltf-transform.donmccurdy.com/classes/core.animation.html), [AnimationChannel](https://gltf-transform.donmccurdy.com/classes/core.animationchannel.html), and [AnimationSampler](https://gltf-transform.donmccurdy.com/classes/core.animationsampler.html) classes. The general approach would be to find the matching AnimationChannel objects affecting the same bone and property in each file (e.g. "Hip" + "rotation"), and then concatenating the "input" and "output" samplers for each so that you have a single resulting AnimationChannel. Since each set of "input" times will start at zero, you would need to increase each time by an offset (duration of prior animations) when adding it to the joined animation channel.

**Reply by kalthwaini** [reply]
Clear. I think I do struggle on the the loaded file structure. I mean the path to the animation channels and samplers starting for the loaded files. If there is a documentation for this it will be great!
